atlasdb-commons/src/main/java/com/palantir/common/proxy/InterruptibleProxy.java

```diff
@@ -35,23 +35,28 @@
  *
  */
 public final class InterruptibleProxy implements DelegatingInvocationHandler {
+    private static final ExecutorService defaultExecutor = PTExecutors.newCachedThreadPool(
+            new NamedThreadFactory("Interruptible Proxy", true /* isDaemon */ ));
```

Comment:
Wondering if we should lazily create this as it really shouldn't be used directly, was thinking along lines:

```
    private static final Supplier<ExecutorService> defaultExecutor = Suppliers.memoize(
            () -> PTExecutors.newCachedThreadPool(new ThreadFactoryBuilder()
                    .setNameFormat("Interruptible-Proxy-%d")
                    .setDaemon(true)
                    .build()));
```